cmd/cluster-agent/api/server.go

```diff
@@ -134,7 +133,7 @@ func StartServer(ctx context.Context, w workloadmeta.Component, taggerComp tagge
 
 	grpcSrv := grpc.NewServer(opts...)
 	pb.RegisterAgentSecureServer(grpcSrv, &serverSecure{
-		taggerServer: taggerserver.NewServer(taggerComp, maxMessageSize),
+		taggerServer: taggerserver.NewServer(taggerComp, maxMessageSize/2),
```

Comment:
this looks like a magic number could you add a comment to explains why you divide by 2 the max message Size here.